### ISSUE TITLE
set_cursor_state(CursorState::Grab) not working correctly on Windows 8.1

### ISSUE THREAD

**Issue by Boscop** [tomaka/glutin#928]
When I call `display.gl_window().window().set_cursor_state(CursorState::Grab)` every frame, at first it works, the cursor is hidden and can't be moved out of the window with the mouse.
But I can still ALT+TAB out of the Window, and then, when I move the cursor back over the window, it's not being grabbed even though I call that function every frame!
After that it behaves like `Hide`, i.e. the cursor is hidden while over the window, but it's not grabbed.

**Comment by andrewtc** [comment]
**TL;DR:** The behavior you describe is actually the expected behavior.

In Windows, there is always only one active window, and that window is the **only** one that has control over the mouse cursor. For example, if some other process (e.g. Chrome, Notepad++) is active and you hover over the window of your own program, the cursor **won't** be grabbed because your window doesn't have permission to manipulate the cursor. The other process does. If you click into your window, however, it will become the active window and the mouse will be grabbed again.

**NOTE:** This behavior isn't something that you can really change. Windows always gives control of the mouse cursor to only one window so that multiple processes aren't fighting over it all the time.

As for the cursor not being grabbed when you ALT+TAB out of your program, Windows purposefully gives you the ability to do this even when the cursor is being constrained, so that you don't get stuck inside some window with no way of getting out. As a user, you have the comfort of knowing that even if a program grabs your cursor and then hangs, you can always ALT+TAB out and kill the process with Task Manager. You wouldn't be able to do this unless switching out of the window also freed up the cursor.

**Comment by francesca64** [comment]
This issue should be reported to [winit](https://github.com/tomaka/winit) instead.
